ci/run-tests.py

```diff
@@ -68,7 +68,11 @@ def change_setting(content, setting, value):
 if args.headless:
     init_contents = change_setting(init_contents, 'PRINT_MODE', 'TEXT')
 
-test_init_file = 'dfhackzzz_test.init'  # Core sorts these alphabetically
+try:
+    os.mkdir('dfhack-config/init')
+except OSError as error:
+    pass
+test_init_file = 'dfhack-config/init/dfhackzzz_test.init'  # Core sorts these alphabetically
```

Comment:
I've been scratching my head on ways to get the old run-tests.py to work with the new DFHack core. I'm not sure there is a good way after all.

However, it might be nice if the new run-tests.py could work with the old DFHack by putting its `.init` file in the right spot. That would enable upgrading the Buildmaster image to allow CI to pass on this PR without failing/hanging on older branches. I think we could do this by installing some sort of "properties" file into `hack` that run-tests.py can read at runtime to figure out where to install a .init file. JSON is pretty easy to parse from Python, and probably easy to generate, although all we would need now is something hardcoded like `{"init_path": "./hack/init"}` (and run-tests could default to `".'` if the file/key isn't present).

Reply:
Modified to check which version of DFHack we're building and place the init script accordingly